### PR TITLE
gh-107298: Fix C API datetime documentation

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
* Issue: gh-107298

----
:books: Documentation preview :books:: https://cpython-previews--108034.org.readthedocs.build/

### CODE REVIEW

File: Doc/c-api/datetime.rst [hunk 1]
```diff
@@ -28,7 +71,7 @@ Type-check macros:
 .. c:function:: int PyDate_Check(PyObject *ob)
 
    Return true if *ob* is of type :c:data:`PyDateTime_DateType` or a subtype of
-   :c:data:`PyDateTime_DateType`.  *ob* must not be ``NULL``.  This function always
+   :c:data:`!PyDateTime_DateType`.  *ob* must not be ``NULL``.  This function always
```
Comment (vstinner) [inline Doc/c-api/datetime.rst:74]: Remark: You could keep the two links to format them the same way. Same remark below.

File: Doc/c-api/datetime.rst [hunk 2]
```diff
@@ -141,7 +184,7 @@ Macros to create objects:
    .. versionadded:: 3.7
 
 
-.. c:function:: PyObject* PyTimeZone_FromOffsetAndName(PyDateTime_DeltaType* offset, PyUnicode* name)
+.. c:function:: PyObject* PyTimeZone_FromOffsetAndName(PyDateTime_DeltaType *offset, PyObject *name)
```
Comment (vstinner) [inline Doc/c-api/datetime.rst]: The first argument type is ``PyObject*``, no?

PyDateTime_CAPI:

```
PyObject *(*TimeZone_FromTimeZone)(PyObject *offset, PyObject *name);
```

Same remark for PyTimeZone_FromOffset().

**Review by vstinner — APPROVED** [review]
LGTM. Feel free to ignore my remarks (or address them in a separated PR).

File: Doc/c-api/datetime.rst (hunk 1 above)
Comment (serhiy-storchaka) [reply Doc/c-api/datetime.rst:74]: They are formatted the same way, but only the first one creates a link. I do not know where it is specified, but the style guide is that you should avoid repeating the same link in one paragraph (or even in few short paragraphs).

File: Doc/c-api/datetime.rst (hunk 2 above)
Comment (serhiy-storchaka) [reply Doc/c-api/datetime.rst]: Good catch.

**Comment by serhiy-storchaka** [comment]
Thank you for your review @vstinner.

**Pull request merged**